spec/unit/provider/package/dpkg_spec.rb

```diff
@@ -52,8 +51,8 @@
   end
 
   before(:each) do
-    allow(provider).to receive(:shell_out!).with("dpkg-deb -W #{new_resource.source}", timeout: 900).and_return(dpkg_deb_status)
-    allow(provider).to receive(:shell_out).with("dpkg -s #{package}", timeout: 900).and_return(dpkg_s_status)
+    allow(provider).to receive(:shell_out!).with("dpkg-deb -W #{source}", timeout: 900).and_return(dpkg_deb_status)
```

Comment:
Ok, found what I was looking for, here: https://github.com/chef/chef/blob/de1f684f415faa54599c6b3abbe211d64a319aa6/spec/unit/provider/package/rpm_spec.rb#L43-L49

I think the case I was interested in was that in some example groups, `shell_out` would be stubbed, and in others it would not. In the example I linked, that was also true for `File.exists?`. Which meant that you could accidentally leave out the stubs for `File.exists?` when the file didn't exist or when a command would fail and the tests would pass until you ran them on a different platform or added a conflicting stub. Anyway, that's not relevant to this code since we're putting at least one method stub at the top level.